test-sbt/shared/src/main/scala/zio/test/runner/ZTestRunner.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test.runner
+
+import com.github.ghik.silencer.silent
+import sbt.testing._
+import zio.internal.{ Platform, PlatformLive }
+import zio.{ RIO, Runtime, ZIO }
+import zio.test._
+import zio.test.runner.ExecutedSpecStructure.Stats
+
+final class ZTestRunner(val args: Array[String], val remoteArgs: Array[String], testClassLoader: ClassLoader)
+    extends Runner {
+  def done(): String = "Done"
+  def tasks(defs: Array[TaskDef]): Array[Task] =
+    defs.map(new ZTestTask(_, testClassLoader))
+}
+
+class ZTestTask(val taskDef: TaskDef, testClassLoader: ClassLoader) extends Task {
+
+  private val platform: Platform = PlatformLive.makeDefault().withReportFailure(_ => ())
+
+  override def tags(): Array[String] = Array.empty
+
+  private def loadSpec[R, L]: RunnableSpec[R, L] = {
+    import org.portablescala.reflect._
+    val fqn = taskDef.fullyQualifiedName.stripSuffix("$") + "$"
+    Reflect
+      .lookupLoadableModuleClass(fqn, testClassLoader)
+      .getOrElse(throw new ClassNotFoundException("failed to load object: " + fqn))
+      .loadModule()
+      .asInstanceOf[RunnableSpec[R, L]]
+  }
+
+  override def execute(eventHandler: EventHandler, loggers: Array[Logger]): Array[Task] = {
+
+    val spec = loadSpec[Any, Any]
+    val handle = for {
+      result <- spec.run
```

Comment:
Try this instead:

```scala
spec.run.flatMap(_.foldM(...))
```
where `foldM` will handle each suite / test. Will clean up a lot of code and eliminate the need for the intermediate structures. 

Let me know if you need a hand or if I've missed something!